### PR TITLE
product creation

### PR DESCRIPTION
**Pull request by YasiucheniaAndrey**

### CODE REVIEW

File: mysite/models.py [hunk 1]
```diff
@@ -1,5 +1,36 @@
 from django.db import models
 
 
+class Product(models.Model):
+    description = models.TextField()
+    short_description = models.CharField(max_length=50)
+    price = models.FloatField()
```
Comment (revimi) [inline mysite/models.py]: Float can't be used for prices

File: mysite/models.py [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
 
-# Create your models here.
+
+class Brand(models.Model):
+    name = models.CharField(max_length=50)
+    description = models.TextField()
+
+
+class Category(models.Model):
+    category = models.CharField(max_length=50)
```
Comment (revimi) [inline mysite/models.py]:
```suggestion
    name = models.CharField(max_length=50)
```

File: mysite/models.py [hunk 3]
```diff
@@ -1,5 +1,36 @@
 from django.db import models
 
 
+class Product(models.Model):
+    description = models.TextField()
+    short_description = models.CharField(max_length=50)
+    price = models.FloatField()
+    category = models.IntegerField()
+    brand = models.IntegerField()
```
Comment (revimi) [inline mysite/models.py]: Use Foreign Key field

File: mysite/models.py [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
+    name = models.CharField(max_length=50)
+    description = models.TextField()
+
+
+class Category(models.Model):
+    category = models.CharField(max_length=50)
+
+
+class Review(models.Model):
+    product_id = models.IntegerField()
```
Comment (revimi) [inline mysite/models.py]: Use Foreign Key field

File: mysite/models.py [hunk 5]
```diff
@@ -1,5 +1,36 @@
...
+    revie_text = models.TextField()
```
Comment (revimi) [inline mysite/models.py]:
```suggestion
    text = models.TextField()
```

File: mysite/models.py [hunk 6]
```diff
@@ -1,5 +1,36 @@
...
+class Brand(models.Model):
+    name = models.CharField(max_length=50)
+    description = models.TextField()
+
+
+class Category(models.Model):
+    category = models.CharField(max_length=50)
+
+
+class Review(models.Model):
```
Comment (revimi) [inline mysite/models.py]: - add `author` field (FK to user)
- add `rating` field (small int 1-5)
- add `timestamp` field

File: mysite/models.py [hunk 7]
```diff
@@ -1,5 +1,36 @@
 from django.db import models
 
 
+class Product(models.Model):
```
Comment (revimi) [inline mysite/models.py]: NTH: add `is_present` field or method

File: mysite/models.py [hunk 8]
```diff
@@ -1,5 +1,36 @@
...
+class Category(models.Model):
+    category = models.CharField(max_length=50)
+
+
+class Review(models.Model):
+    product_id = models.IntegerField()
+    revie_text = models.TextField()
+
+
+class User(models.Model):
```
Comment (revimi) [inline mysite/models.py]: Use standard Django user model

Comment (revimi) [reply mysite/models.py]: https://www.google.com/search?q=django+user+model&oq=django+user+mode&aqs=chrome.0.0i512j69i57j0i20i263i512j0i512l7.48327j0j4&sourceid=chrome&ie=UTF-8

File: mysite/models.py [hunk 9]
```diff
@@ -1,5 +1,36 @@
...
+    login_name = models.CharField(max_length=50)
+    id_review = models.IntegerField()
+    id_product_in_order = models.IntegerField()
```
Comment (revimi) [inline mysite/models.py]: ???

File: mysite/models.py [hunk 10]
```diff
@@ -1,5 +1,36 @@
...
+
+
+class Order(models.Model):
+    product_id = models.IntegerField()
```
Comment (revimi) [inline mysite/models.py]: FK

File: mysite/models.py [hunk 11]
```diff
@@ -1,5 +1,36 @@
...
+    user_id = models.IntegerField()
```
Comment (revimi) [inline mysite/models.py]: FK

File: mysite/models.py [hunk 12]
```diff
@@ -1,5 +1,36 @@
...
+    is_paid = models.BooleanField()
```
Comment (revimi) [inline mysite/models.py]: https://docs.djangoproject.com/en/4.1/ref/models/fields/#choices
```suggestion
    status = models.CharField()
```

File: mysite/models.py [hunk 13]
```diff
@@ -1,5 +1,36 @@
...
+    order_date = models.DateTimeField()
```
Comment (revimi) [inline mysite/models.py]:
```suggestion
    timestamp = models.DateTimeField()
    due_date = models.DateTimeField() (Optional)
```

File: mysite/models.py [hunk 14]
```diff
@@ -1,5 +1,36 @@
 from django.db import models
```
Comment (revimi) [inline mysite/models.py:1]: Add migrations to the project

File: .idea/vcs.xml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (revimi) [inline .idea/vcs.xml]: add `.idea` to `.gitignore`

**Review by revimi — CHANGES_REQUESTED** [review]

File: mysite/models.py (hunk 1 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: using DecimalField

File: mysite/models.py (hunk 2 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 3 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 4 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 5 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 6 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 7 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 10 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 11 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 12 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: .idea/vcs.xml (hunk 15 above)
Comment (YasiucheniaAndrey) [reply .idea/vcs.xml]: done

File: mysite/models.py (hunk 14 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py:1]: done

File: mysite/models.py (hunk 13 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 8 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: from django.conf import settings

user_id = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        on_delete=models.CASCADE
    )
как я понял нужно так

File: mysite/models.py (hunk 9 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py [hunk 16]
```diff
@@ -1,5 +1,62 @@
...
+    )
+    STATUS = [
+        ('not paid', 'order is not paid yet '),
+        ('paid', 'order is paid')
+    ]
+    status = models.CharField(
+        max_length=8,
+        choices=STATUS,
+        default = 'not paid'
+    )
```
Comment (revimi) [inline mysite/models.py]: Let's have constants for choices
```suggestion
    PAID = "paid"
    NOT_PAID = "not_paid"
    STATUS = [
        (PAID, 'paid'),
        (NOT_PAID, 'not paid')
    ]
    status = models.CharField(
        max_length=8,
        choices=STATUS,
        default=NOT_PAID
    )
```

File: mysite/models.py [hunk 17]
```diff
@@ -1,5 +1,62 @@
...
+
+class Category(models.Model):
+    name = models.CharField(max_length=50)
+
+
+class Review(models.Model):
+    author = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        default=""
```
Comment (revimi) [inline mysite/models.py]: It can't be blank. We can't have a default for the author field

File: mysite/models.py [hunk 18]
```diff
@@ -1,5 +1,62 @@
 from django.db import models
```
Comment (revimi) [inline mysite/models.py:1]: Also, let's have separate applications for Product and Order

**Review by revimi — CHANGES_REQUESTED** [review]

File: mysite/models.py (hunk 17 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: mysite/models.py (hunk 16 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py]: done

File: order/models.py [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+    PAID = 'order is paid'
+    NOT_PAID = 'order is not paid'
+    STATUS = [
+        (PAID, 'order is not paid yet '),
```
Comment (revimi) [inline order/models.py]: paid or not paid?

File: order/models.py [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+    product = models.ForeignKey(
+        Product,
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+    PAID = 'order is paid'
+    NOT_PAID = 'order is not paid'
```
Comment (revimi) [inline order/models.py]: It is something we are going to store in the database. Let's use short strings with underscores
```suggestion
    PAID = 'paid'
    NOT_PAID = 'not_paid'
```

File: .idea/dataSources.xml [hunk 21]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (revimi) [inline .idea/dataSources.xml]: Please completely remove `.idea` folder

File: mysite/models.py (hunk 18 above)
Comment (YasiucheniaAndrey) [reply mysite/models.py:1]: done

File: order/models.py (hunk 19 above)
Comment (YasiucheniaAndrey) [reply order/models.py]: PAID

File: order/models.py (hunk 20 above)
Comment (YasiucheniaAndrey) [reply order/models.py]: done

File: order/models.py [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+        (PAID, 'order is paid '),
+        (NOT_PAID, 'order is not paid')
+    ]
+    status = models.CharField(
+        max_length=20,
+        choices=STATUS,
+        default = NOT_PAID
+    )
+    timestamp = models.DateTimeField()
+# Create your models here.
```
Comment (revimi) [inline order/models.py]: Please remove the comment

File: .gitignore [hunk 23]
```diff
@@ -0,0 +1,2 @@
+.idea
+venv
```
Comment (revimi) [inline .gitignore]: Please add a blank line in the end of the file

File: order/admin.py [hunk 24]
```diff
@@ -0,0 +1,6 @@
+from django.contrib import admin
+from .models import Order
+
+
+admin.site.register(Order)
+# Register your models here.
```
Comment (revimi) [inline order/admin.py]: Please remove the comment

File: product/admin.py [hunk 25]
```diff
@@ -0,0 +1,7 @@
+from django.contrib import admin
+from .models import Product, Review
+
+admin.site.register(Product)
+admin.site.register(Review)
+
+# Register your models here.
```
Comment (revimi) [inline product/admin.py]: Please remove the comment

**Review by revimi — APPROVED** [review]

File: order/models.py [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+    product = models.ForeignKey(
+        Product,
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+    PAID = 'paid'
+    NOT_PAID = 'ssnot paid'
```
Comment (revimi) [inline order/models.py]:
```suggestion
    NOT_PAID = 'not_paid'
```

File: .gitignore (hunk 23 above)
Comment (YasiucheniaAndrey) [reply .gitignore]: done

File: .idea/dataSources.xml (hunk 21 above)
Comment (YasiucheniaAndrey) [reply .idea/dataSources.xml]: done

File: order/models.py (hunk 22 above)
Comment (YasiucheniaAndrey) [reply order/models.py]: done

File: order/admin.py (hunk 24 above)
Comment (YasiucheniaAndrey) [reply order/admin.py]: done

File: product/admin.py (hunk 25 above)
Comment (YasiucheniaAndrey) [reply product/admin.py]: done

File: order/models.py (hunk 26 above)
Comment (YasiucheniaAndrey) [reply order/models.py]: done

**Pull request merged**